Sources/SwiftGenKit/Parsers/Strings/FileTypeParser/StringsFileParser.swift

```diff
@@ -23,100 +23,96 @@ extension Strings {
       }
 
-      let entries = try PropertyListDecoder()
+      var entries = try PropertyListDecoder()
         .decode([String: String].self, from: data)
-        .map { key, translation -> (String, Entry) in
-          let entry = try Entry(key: key, translation: translation, keyStructureSeparator: options[Option.separator])
-          return (key, entry)
+        .map { key, translation -> Entry in
+          try Entry(key: key, translation: translation, keyStructureSeparator: options[Option.separator])
         }
 
-      var dict = Dictionary(uniqueKeysWithValues: entries)
+      parseComments(path: path, entries: &entries)
 
-      if let string: String = try? path.read() {
-        let scanner = Scanner(string: string)
-        while !scanner.isAtEnd {
-          let comment = scanner.scanComment()
-          if let key = scanner.scanKey() {
-            dict[key]?.comment = comment
-          }
-        }
+      return entries
+    }
+
+    /// Try to parse the comments in a strings file, and add them to the corresponding entries.
+    ///
+    /// First we remove all keys & values, and replace them by "magic markers", so that we can more easily parse the
+    /// comments. Then we try to parse a comment, and then a marker, and repeat this until we're done.
+    func parseComments(path: Path, entries: inout [Entry]) {
+      guard var contents: String = try? path.read() else { return }
+
+      // replace entries with markers
+      for (index, entry) in zip(0..., entries) {
+        let key = entry.key.escapedStringsRegex
+        let translation = entry.translation.escapedStringsRegex
+        contents = contents.replacingOccurrences(
+          of: #"(?m)"\#(key)"\s*=\s*"\#(translation)"\s*;"#,
+          with: "\(String.startMarker)\(index)\(String.endMarker)",
+          options: .regularExpression
+        )
       }
 
-      return Array(dict.values)
+      // parse comment and marker in turns
+      let scanner = Scanner(string: contents)
+      while !scanner.isAtEnd {
+        let comment = scanner.scanComment()
+        if let index = scanner.scanEntryIndex() {
+          entries[index].comment = comment
+        }
+      }
     }
   }
 }
 
 private extension String {
-  static var startComment: String { "/*" }
-  static var endComment: String { "*/" }
-  static var quote: String { "\"" }
+  static let startComment = "/*"
+  static let endComment = "*/"
+  static let startMarker = "___SWIFTGEN___MAGIC___MARKER___"
+  static let endMarker = ";"
+
+  /// Generates a regex for matching a key or value in a strings file
+  ///
+  /// Note that keys & values need to escape all `"\n\r\`. On top of that, newlines need to be prefixed with a `\`.
+  var escapedStringsRegex: String {
+    NSRegularExpression.escapedPattern(
+      for: self.replacingOccurrences(of: "\"", with: #"\""#)
+        .replacingOccurrences(of: "\r", with: #"\\#r"#)
+        .replacingOccurrences(of: "\n", with: #"\\#n"#)
+    )
+  }
```

Comment:
String files support both newlines and escaped newlines. So this may not correctly match the contents of the string file if it's using non-escaped newlines.

Based on `parseQuotedPlistString` in
https://opensource.apple.com/source/CF/CF-550/CFPropertyList.c.auto.html

For example, the following keys will be parsed to have identical values.

```
let string = """
"key1" = "value
value";
"key2" = "value\nvalue";
"""

let decoder = PropertyListDecoder()
let dict = try decoder.decode([String: String].self, from: Data(string.utf8))

dict["key1"] == dict["key2"] // true
```